src/Compilers/CSharp/Test/Symbol/Symbols/RequiredMembersTests.cs

```diff
@@ -1222,4 +1222,1388 @@ struct S3
+            // (12,8): error CS9509: The required members list for the base type 'Derived' is malformed and cannot be interpreted. To use this constructor, apply the `SetsRequiredMembers` attribute.
+            // record DerivedDerived2 : Derived
+            Diagnostic(ErrorCode.ERR_RequiredMembersBaseTypeInvalid, "DerivedDerived2").WithArguments("Derived").WithLocation(12, 8),
+            // (12,8): error CS9508: The required members list for 'Derived' is malformed and cannot be interpreted.
+            // record DerivedDerived2 : Derived
+            Diagnostic(ErrorCode.ERR_RequiredMembersInvalid, "DerivedDerived2").WithArguments("Derived").WithLocation(12, 8),
+            // (12,8): error CS9508: The required members list for 'Derived' is malformed and cannot be interpreted.
+            // record DerivedDerived2 : Derived
+            Diagnostic(ErrorCode.ERR_RequiredMembersInvalid, "DerivedDerived2").WithArguments("Derived").WithLocation(12, 8),
+            // (15,8): error CS9509: The required members list for the base type 'Derived' is malformed and cannot be interpreted. To use this constructor, apply the `SetsRequiredMembers` attribute.
+            // record DerivedDerived3() : Derived;
+            Diagnostic(ErrorCode.ERR_RequiredMembersBaseTypeInvalid, "DerivedDerived3").WithArguments("Derived").WithLocation(15, 8),
+            // (15,8): error CS9509: The required members list for the base type 'Derived' is malformed and cannot be interpreted. To use this constructor, apply the `SetsRequiredMembers` attribute.
+            // record DerivedDerived3() : Derived;
+            Diagnostic(ErrorCode.ERR_RequiredMembersBaseTypeInvalid, "DerivedDerived3").WithArguments("Derived").WithLocation(15, 8),
+            // (15,8): error CS9508: The required members list for 'Derived' is malformed and cannot be interpreted.
+            // record DerivedDerived3() : Derived;
+            Diagnostic(ErrorCode.ERR_RequiredMembersInvalid, "DerivedDerived3").WithArguments("Derived").WithLocation(15, 8),
+            // (15,8): error CS9508: The required members list for 'Derived' is malformed and cannot be interpreted.
+            // record DerivedDerived3() : Derived;
+            Diagnostic(ErrorCode.ERR_RequiredMembersInvalid, "DerivedDerived3").WithArguments("Derived").WithLocation(15, 8)
+        );
+    }
+
+    [Fact]
+    public void EnforcedRequiredMembers_ShadowedInSource_06()
+    {
+        var original = @"
+public record Base
+{
+    public required int P { get; init; }
+}
+";
+
+        var originalComp = CreateCompilationWithRequiredMembers(new[] { original, IsExternalInitTypeDefinition }, assemblyName: "original");
+        CompileAndVerify(originalComp).VerifyDiagnostics();
+
+        // This IL is the equivalent of:
+        // public record Derived : Base
+        // {
+        //    public new required int P { get; init; }
+        // }
+
+        var ilSource = @"
+.assembly extern original {}
```

Comment:
Do we need to use a record to meaningfully test the scenario here? Seems like IL would be much smaller without the record members.

Reply:
Yes. The record is the point of this test.